src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -7922,15 +7818,15 @@ def fillna(
         BaseQueryCompiler
             New QueryCompiler with all null values filled.
         """
-        default_class = SeriesDefault if self_is_series else DataFrameDefault
-        fallback_func = (
-            native_pd.Series.fillna if self_is_series else native_pd.DataFrame.fillna
-        )
-        use_fallback = False
-        if limit or downcast:
-            # fallback before having parallel implementation
-            # TODO: SNOW-891788 support limit
-            use_fallback = True
+        # TODO: SNOW-891788 support limit
+        if limit:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas fillna API doesn't yet support 'limit' parameter"
```

Comment:
In cases, where some pandas APIs are considered synonyms of other pandas APIs (like ffill/bfill and fillna in this case), we may want to pass the original API name to show in the error message. Not critical though.